working_with_apis.md:  Replace http with https for openweathermap api

fbuys opened the pull request:
<!-- Thank you for taking the time to contribute to The Odin Project. In order to get this pull request (PR) merged in a reasonable amount of time, you must complete this entire template. -->

## Because
<!-- Summarize the purpose or reasons for this PR, e.g. what problem it solves or what benefit it provides. -->

If we use `http` instead of `https` github pages throws an error.

## This PR
<!-- A bullet point list of one or more items describing the specific changes. -->

- Replaced instances of `http` with `https`

## Issue
<!--
If this PR closes an open issue in this repo, replace the XXXXX below with the issue number, e.g. Closes #2013.

If this PR closes an open issue in another TOP repo, replace the #XXXXX with the URL of the issue, e.g. Closes https://github.com/TheOdinProject/curriculum/issues/XXXXX

If this PR does not close, but is related to another issue or PR, you can link it as above without the 'Closes' keyword, e.g. 'Related to #2013'.
-->
Closes https://github.com/TheOdinProject/curriculum/issues/24687


## Pull Request Requirements
<!-- Replace the whitespace between the square brackets with an 'x', e.g. [x]. After you create the PR, they will become checkboxes that you can click on. -->
-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [ ] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

01zulfi reviewed (commented):
> If we use http instead of https github pages throws an error.

Can you elaborate on what error GitHub Pages throws?

fbuys commented:
> > If we use http instead of https github pages throws an error.
> 
> Can you elaborate on what error GitHub Pages throws?

Sorry I did not replicate this issue, instead I addressed feedback from [this issue](https://github.com/TheOdinProject/curriculum/issues/24687).

I can try to replicate the issue so I can confirm the error message we get.

01zulfi reviewed (approved):
> Sorry I did not replicate this issue, instead I addressed feedback from https://github.com/TheOdinProject/curriculum/issues/24687.
> 
> I can try to replicate the issue so I can confirm the error message we get.

It's fine. I tried accessing the API, I get this: 
![image](https://user-images.githubusercontent.com/85733202/199975261-d1330e54-5af6-4cc8-9a47-c3005cf6871d.png)

The error message has a typo in the lesson which this PR will fix. Additionally, `https` will remove that "not secure" chip from the browser tab.
